internal/engine/worker/federated/activitypub/mastodon/worker.go

```diff
@@ -41,19 +48,307 @@ func (w *worker) Tags() []tag.Tag {
 }
 
 func (w *worker) Types() []schema.Type {
-	return []schema.Type{}
+	return []schema.Type{
+		typex.SocialComment,
+		typex.SocialPost,
+		typex.SocialShare,
+	}
 }
 
 // Filter returns a source filter.
 func (w *worker) Filter() engine.DataSourceFilter {
 	return nil
 }
 
-func (w *worker) Transform(_ context.Context, _ engine.Task) (*activityx.Activity, error) {
+// Transform processes the task and converts it into an activity.
+func (w *worker) Transform(ctx context.Context, task engine.Task) (*activityx.Activity, error) {
 	zap.L().Info("[mastodon/worker.go] reached Transform()")
-	return nil, nil
+
+	activityPubTask, ok := task.(*source.Task)
+
+	if !ok {
+		return nil, fmt.Errorf("invalid task type: %T", task)
+	}
+
+	activity, err := task.BuildActivity(activityx.WithActivityPlatform(w.Platform()))
+
+	if err != nil {
+		return nil, fmt.Errorf("build activity: %w", err)
+	}
+
+	// Handle ActivityPub message.
+	switch activityPubTask.Message.Type {
+	case "Create":
+		err := w.handleActivityPubCreate(ctx, activityPubTask.Message, activity)
+		if err != nil {
+			return nil, fmt.Errorf("error occured in handleActivityPubCreate")
+		}
+	case "Announce":
```

Comment:
```suggestion
using an enumeration is better.
```

Reply:
defined message types in mastodon provider directory and used enumeration here